TheJumpCloud/jumpcloud-ADMU
# Sa 3471 admu disable scheduled tasks

## Issues
* [SA-3471](https://jumpcloud.atlassian.net/browse/<jira-id>) - SA-3471-disable-scheduled-stasks

## What does this solve?
Some scheduled tasks prevents migrations from occuring. This change disables tasks that are on "Ready" state before the migration to ensure migration runs without tasks preventing it.
## Is there anything particularly tricky?
Checking if the tasks are disabled in the middle of the migration
## How should this be tested?
1. Open two PWSH terminal, for migration and task check
2. Run this command to check the pre-migration tasks that are not Windows and Ready: `Get-ScheduledTask | Where-Object { $_.TaskPath -notlike "*\Microsoft\Windows*" -and $_.State -ne "Disabled" -and $_.state -ne "Running" }`
3. Run ADMu
4. In the middle of migration, check if the tasks from pre-migration check are disabled: `Get-ScheduledTask | Where-Object { $_.TaskPath -notlike "*\Microsoft\Windows*" -and $_.State -eq "Disabled" }`
5. Lastly, after the migration, check if pre-migration tasks are set to "Ready": `Get-ScheduledTask | Where-Object { $_.TaskPath -notlike "*\Microsoft\Windows*" -and $_.State -ne "Disabled" -and $_.state -ne "Running" }`

## Screenshots
Pre:
<img width="960" alt="image" src="https://github.com/TheJumpCloud/jumpcloud-ADMU/assets/97972790/8e760ccb-cb06-458c-b5c5-fddac3d7378d">

Mid and after migration:
<img width="960" alt="image" src="https://github.com/TheJumpCloud/jumpcloud-ADMU/assets/97972790/0f1078bf-27b9-478a-b909-43f2ef97c790">



[SA-3471]: https://jumpcloud.atlassian.net/browse/SA-3471?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
